### ISSUE TITLE
tests not running with phantom nor travis

### ISSUE THREAD

**Issue by anthonymayer** [Sanjo/meteor-jasmine#115]
I'm trying to get tests running on phantomjs to work on travis, but the tests aren't running. When I just tried running it on travis I get https://travis-ci.org/anthonymayer/Telescope/builds/45439535, where the test just never finishes. I tried it locally with JASMINE_BROWSER=PhantomJS and it also never runs the tests. Here's the output from running it with velocity debug on (ignore all the 34ml's):

```
JASMINE_BROWSER=PhantomJS VELOCITY_DEBUG=1 VELOCITY_DEBUG_MIRROR=1 VELOCITY_SINGLE_RUN=1 meteor --test
[[[[[ ~/Telescope ]]]]]

=> Started proxy.
=> Started MongoDB.
[34mI20141230-18:04:33.495(1)? [39m[MailChimp] Error: MailChimp Options have not been set in your settings.json file.
[34mI20141230-18:04:34.089(1)? [39m[velocity] adding velocity core
[34mI20141230-18:04:34.128(1)? [39m[velocity] Not adding mirror-registrar because NODE_ENV is development and IS_MIRROR is false
[34mI20141230-18:04:34.414(1)? [39m[velocity] Velocity not started. Queueing func
[34mI20141230-18:04:34.476(1)? [39m[node-soft-mirror] adding server code
[34mI20141230-18:04:34.477(1)? [39m[velocity] Velocity not started. Queueing func
[34mW20141230-18:04:34.527(1)? (STDERR) [39m[35m[AccountsTemplates] WARNING: showForgotPasswordLink set to true, but MAIL_URL is not configured![39m
[34mI20141230-18:04:34.535(1)? [39mStarting artwells:queue for Herald
[34mI20141230-18:04:34.535(1)? [39m[velocity] Server startup
[34mI20141230-18:04:34.535(1)? [39m[velocity] app dir /Users/anthonymayer/Telescope
[34mI20141230-18:04:34.536(1)? [39m[velocity] config = {
[34mI20141230-18:04:34.536(1)? [39m  "jasmine-client-integration": {
[34mI20141230-18:04:34.536(1)? [39m    "regex": "^tests/jasmine/client/integration/.+\\.(js|coffee|litcoffee|coffee\\.md)$",
[34mI20141230-18:04:34.536(1)? [39m    "name": "jasmine-client-integration",
[34mI20141230-18:04:34.536(1)? [39m    "_regexp": {}
[34mI20141230-18:04:34.536(1)? [39m  },
[34mI20141230-18:04:34.537(1)? [39m  "jasmine-client-unit": {
[34mI20141230-18:04:34.537(1)? [39m    "regex": "^tests/jasmine/client/unit/.+\\.(js|coffee|litcoffee|coffee\\.md)$",
[34mI20141230-18:04:34.537(1)? [39m    "name": "jasmine-client-unit",
[34mI20141230-18:04:34.537(1)? [39m    "_regexp": {}
[34mI20141230-18:04:34.537(1)? [39m  },
[34mI20141230-18:04:34.537(1)? [39m  "jasmine-server-integration": {
[34mI20141230-18:04:34.537(1)? [39m    "regex": "^tests/jasmine/server/integration/.+\\.(js|coffee|litcoffee|coffee\\.md)$",
[34mI20141230-18:04:34.537(1)? [39m    "name": "jasmine-server-integration",
[34mI20141230-18:04:34.538(1)? [39m    "_regexp": {}
[34mI20141230-18:04:34.538(1)? [39m  },
[34mI20141230-18:04:34.538(1)? [39m  "jasmine-server-unit": {
[34mI20141230-18:04:34.538(1)? [39m    "regex": "^tests/jasmine/server/unit/.+\\.(js|coffee|litcoffee|coffee\\.md)$",
[34mI20141230-18:04:34.538(1)? [39m    "name": "jasmine-server-unit",
[34mI20141230-18:04:34.538(1)? [39m    "_regexp": {}
[34mI20141230-18:04:34.538(1)? [39m  }
[34mI20141230-18:04:34.539(1)? [39m}
[34mI20141230-18:04:34.539(1)? [39m[velocity] resetting the world
[34mI20141230-18:04:34.539(1)? [39m[velocity] frameworks with disable auto reset: []
[34mI20141230-18:04:34.597(1)? [39m[velocity] No framework registered for tests/jasmine/fixtures/test-fixture-ignore.js
[34mI20141230-18:04:34.603(1)? [39m[velocity] Target framework for tests/jasmine/client/unit/navSpec.js is jasmine-client-unit
[34mI20141230-18:04:34.605(1)? [39m[velocity] File scan complete, now watching /tests
[34mI20141230-18:04:34.605(1)? [39m[velocity] Triggering queued startup functions
[34mI20141230-18:04:34.613(1)? [39m[proxy-package-sync] Test file added tests/jasmine/client/unit/navSpec.js
[34mI20141230-18:04:34.622(1)? [39m[sanjo:jasmine]: No tests for jasmine-client-integration found. Reporting completed.
[34mI20141230-18:04:34.629(1)? [39m[velocity-node-mirror] Server restarted.
[34mI20141230-18:04:34.630(1)? [39m[velocity-node-mirror] Aggressively restarting all mirrors
[34mI20141230-18:04:34.630(1)? [39m[node-soft-mirror] Aggressively reload client
[34mI20141230-18:04:34.638(1) (percolatestudio_synced-cron.js:74) [39mSyncedCron: scheduled "scheduleNewsletter" next run @Mon Jan 05 2015 00:00:00 GMT+0100 (CET)
[34mI20141230-18:04:34.642(1) (percolatestudio_synced-cron.js:74) [39mSyncedCron: scheduled "Post by RSS feed" next run @Tue Dec 30 2014 18:30:00 GMT+0100 (CET)
=> Started your app.

=> App running at: http://localhost:3000/
[34mI20141230-18:04:34.718(1)? [39m[node-soft-mirror]: LongRunningChildProcess.constructor(taskName=jasmine-client-unit)
[34mI20141230-18:04:34.719(1)? [39m[node-soft-mirror]: LongRunningChildProcess.readPid()
[34mI20141230-18:04:34.720(1)? [39m[node-soft-mirror]: LongRunningChildProcess.readPid returns 53733
[34mI20141230-18:04:34.720(1)? [39m[node-soft-mirror]: LongRunningChildProcess.isRunning returns false
[34mI20141230-18:04:34.722(1)? [39m[node-soft-mirror]: LongRunningChildProcess.spawn() { command: '/Users/anthonymayer/Telescope/.meteor/local/build/programs/server/npm/sanjo:karma/node_modules/karma/bin/karma',
[34mI20141230-18:04:34.722(1)? [39m  args: 
[34mI20141230-18:04:34.722(1)? [39m   [ 'start',
[34mI20141230-18:04:34.722(1)? [39m     '/Users/anthonymayer/Telescope/.meteor/local/karma/jasmine-client-unit.config.js' ] }
[34mI20141230-18:04:34.723(1)? [39m[node-soft-mirror]: LongRunningChildProcess.isRunning returns false
[34mI20141230-18:04:34.723(1)? [39m[node-soft-mirror]: /Users/anthonymayer/Telescope/.meteor/local/build/programs/server/assets/packages/velocity_node-soft-mirror/lib/spawnScript.js
[34mI20141230-18:04:34.723(1)? [39m[node-soft-mirror]: The pid of the main Meteor app process is 53958
[34mI20141230-18:04:34.723(1)? [39m[node-soft-mirror]: LongRunningChildProcess.spawn is spawning 'karma'
[34mI20141230-18:04:34.726(1)? [39m[node-soft-mirror]: LongRunningChildProcess._setPid(pid=53984)
[34mI20141230-18:04:34.726(1)? [39m[node-soft-mirror]: Saving jasmine-client-unit pid 53984 to /Users/anthonymayer/Telescope/.meteor/local/run/jasmine-client-unit.pid
[34mI20141230-18:04:34.730(1)? [39m[sanjo:jasmine]: No tests for jasmine-server-integration found. Reporting completed.
[34mI20141230-18:04:34.819(1)? [39m[proxy-package-sync] Checking if a new package.js needs to be written because {} undefined
[34mI20141230-18:04:34.819(1)? [39m[proxy-package-sync] Generating in-memory package.js
[34mI20141230-18:04:34.821(1)? [39m[proxy-package-sync] Fixture files list length:  0
[34mI20141230-18:04:34.822(1)? [39m[proxy-package-sync] Test files list length:  1
[34mI20141230-18:04:34.823(1)? [39m[proxy-package-sync] Test file will not be included in mirror tests/jasmine/client/unit/navSpec.js
[34mI20141230-18:04:34.824(1)? [39m[proxy-package-sync] Comparing:
[34mI20141230-18:04:34.824(1)? [39m[proxy-package-sync] - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[34mI20141230-18:04:34.824(1)? [39m[proxy-package-sync] - - - - - - - - - currentPackageJS: - - - - - - - - - - - - -
[34mI20141230-18:04:34.825(1)? [39mPackage.describe({
[34mI20141230-18:04:34.825(1)? [39m	name: "velocity:test-proxy",
[34mI20141230-18:04:34.825(1)? [39m	summary: "Dynamically created package to expose test files to mirrors",
[34mI20141230-18:04:34.825(1)? [39m	version: "0.0.4",
[34mI20141230-18:04:34.825(1)? [39m	debugOnly: true
[34mI20141230-18:04:34.826(1)? [39m});
[34mI20141230-18:04:34.826(1)? [39m
[34mI20141230-18:04:34.826(1)? [39mPackage.on_use(function (api) {
[34mI20141230-18:04:34.826(1)? [39m	api.use("coffeescript", ["client", "server"]);
[34mI20141230-18:04:34.827(1)? [39m});
[34mI20141230-18:04:34.827(1)? [39m[proxy-package-sync] - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[34mI20141230-18:04:34.827(1)? [39m[proxy-package-sync] - - - - - - - - - generatedPackageJsContent - - - - - - - - -
[34mI20141230-18:04:34.827(1)? [39mPackage.describe({
[34mI20141230-18:04:34.827(1)? [39m	name: "velocity:test-proxy",
[34mI20141230-18:04:34.828(1)? [39m	summary: "Dynamically created package to expose test files to mirrors",
[34mI20141230-18:04:34.828(1)? [39m	version: "0.0.4",
[34mI20141230-18:04:34.828(1)? [39m	debugOnly: true
[34mI20141230-18:04:34.828(1)? [39m});
[34mI20141230-18:04:34.828(1)? [39m
[34mI20141230-18:04:34.828(1)? [39mPackage.on_use(function (api) {
[34mI20141230-18:04:34.828(1)? [39m	api.use("coffeescript", ["client", "server"]);
[34mI20141230-18:04:34.829(1)? [39m});
[34mI20141230-18:04:34.829(1)? [39m[proxy-package-sync] - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[34mI20141230-18:04:34.829(1)? [39m[proxy-package-sync] No changes to package.js file required
```

I also tried with a local copy of meteor-jasmine and changed the phantomjsLauncher paths in client/unit/ClientUnitTestFramework.js, since ".meteor/local/build/programs/server/node_modules/.bin/phantomjs" doesn't exist to point at my local copy of phantom (/usr/local/bin/phantomjs), but that didn't change it.

I'm still trying to dig through the karma package chain to see what's happening, but help would be appreciated. The tests definitely run if I just do "meteor --test" so it's definitely an issue with phantom. My guess is that it just never correctly starts phantom, but doesn't register an error either, but I'm not sure.

**Comment by Sanjo** [comment]
Have a look in the `.meteor/local/log/jasmine-client-unit.log` file if an error is reported. This is the output of the Karma process. Doesn't point `.meteor/local/build/programs/server/node_modules/.bin/phantomjs` to the PhantomJS that ships with Meteor in your case?

**Comment by Sanjo** [comment]
I've just tested it locally. I get an error too. The PhantomJS executable is not there. I will see how I can fix this.

**Comment by Sanjo** [comment]
In the next release phantomjs is expected to be in the path (must be installed with `npm install -g phantomjs`).

**Comment by anthonymayer** [comment]
Looks like the actual error I'm getting is "Some of your tests did a full page reload!". I get the same error on SauceLabs too, but not if I run against Chrome locally. Any ideas(or even on how to debug it) on why it works with Chrome but not Phantom or Sauce? Here's the test https://github.com/TelescopeJS/Telescope/pull/662

**Comment by Sanjo** [comment]
I tried it myself and couldn't figure out what reloads the page. But I don't know how to debug in PhantomJS. Couldn't you just use Chrome for running the tests?

**Comment by anthonymayer** [comment]
I want to run the tests on travis so I can't use Chrome. Supposedly Firefox will work so once #116 (I should have a pr ready later today) is resolved I'll try that. This still seems like something that would be nice to resolve especially since it happens on SauceLabs Chrome too. I'll try adding logging to see when it thinks it's reloading and try to figure out why it's different in Chrome and Phantom

**Comment by Sanjo** [comment]
Relevant code in Karma runner: https://github.com/karma-runner/karma/blob/master/client/karma.js#L61.

**Comment by Sanjo** [comment]
iron:router now gets disabled in Client Unit mode. Will be available in the next release (after 0.14.0).
